Sources/PackageManagerDocs/Documentation.docc/Templates.md

```diff
@@ -0,0 +1,458 @@
+
+The executable allows authors to define user-facing interfaces which gather important consumer input needed by the template to run, using Swift Argument Parser for a rich command-line experience with subcommands, options, and flags.
+
+To learn about using a package template, read <doc:CreatingSwiftPackage#Creating-a-Package-based-on-a-custom-template>.
+
+## Writing a Template
+
+The first step when writing a package template is to decide what kind of template you need and what base package structure it should start with. Templates can build off of any kind of Swift package: executables, libraries, plugins, or even empty packages that will be further customized.
+
+### Declaring a template in the package manifest
+
+Like all package components, templates are declared in the package manifest. This is done using a `templateTarget` entry in the `targets` section of the package. Templates must be visible to other packages in order to be ran. Thus, there needs to be a corresponding `template` entry in the `products` section as well:
+
+```swift
+// swift-tools-version: 6.1
+import PackageDescription
+
+let package = Package(
+    name: "MyTemplates",
+    products: [
+        .template(name: "LibraryTemplate"),
+        .template(name: "ExecutableTemplate"),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/apple/swift-argument-parser", from: "1.3.0"),
+    ],
+    targets: [
+        .template(
+            name: "LibraryTemplate",
+            dependencies: [
+                .product(name: "ArgumentParser", package: "swift-argument-parser"),
+            ],
+            initialPackageType: .library,
+            description: "Generate a Swift library package"
+        ),
+        .template(
+            name: "ExecutableTemplate", 
+            dependencies: [
+                .product(name: "ArgumentParser", package: "swift-argument-parser"),
+            ],
+            initialPackageType: .executable,
+            templatePermissions: [
+                .writeToPackageDirectory(reason: "Generate source files and documentation"),
+            ],
+            description: "Generate an executable package with optional features"
+        ),
+    ]
+)
+```
+
+The `templateTarget` declares the name and capability of the template, along with its dependencies. The `initialPackageType` specifies the base package structure that SwiftPM will set up before invoking the template — this can be `.library`, `.executable`, `.tool`, `.buildToolPlugin`, `.commandPlugin`, `.macro`, or `.empty`.
+
+The Swift script files that implement the logic of the template are expected to be in a directory named the same as the template, located under the `Templates` subdirectory of the package. The template also expects Swift script files in a directory with the same name as the template, alongside a `Plugin` suffix, located under the `Plugins` subdirectory of the package.
```

Comment:
```suggestion
The Swift Package Manager expects the Swift script files that implement the logic of the template to be in a directory with the same as the template, located within a `Templates` subdirectory of the package. The Package Manager also expects Swift script files in a directory with the same name as the template, alongside a `Plugin` suffix, located under the `Plugins` subdirectory of the package.
```

This is complex enough that it's a bit hard to parse right off the bat. I'd suggest that we include an example directory structure inline here - such as the output of the `tree` command, illustrating the expectations to make it easier to understand.